### DISCUSSION TITLE
Mime type of faked UploadedFile models not accepted

### DISCUSSION THREAD

**Discussion by maxwoerner** [spatie/laravel-medialibrary#3469] [Ideas]
Hi there,

first of all: thank you for creating this awesome package!

I am trying to create feature tests leveraging the `fake()` method of the `Illuminate\Http\UploadedFile` class. However, I am not able to set a custom mime type like `audio/mpeg` as this is not accepted by the media collection even though its specified in the collection.

I am registering the media collection like this:
```php
public function registerMediaCollections(): void
{
    $this->addMediaCollection('audio')
        ->useDisk('s3')
        ->acceptsMimeTypes([
            'audio/mpeg',
        ]);
}
```

In the test, I am creating a fake `UploadedFile` model with the custom mime type `audio/mpeg` and add it to the `audio` collection:
```php
$uploadedFile = UploadedFile::fake()->create(
    'audio.mp3', 1000, 'audio/mpeg'
);

$this->song
    ->addMedia($uploadedFile)
    ->toMediaCollection('audio');
```

However, this fails with the following error:
```text
The file with properties `name: audio.mp3, size: 0, mime: application/x-empty` was not accepted into the collection named `audio` of model `App\Models\Song` with id `1`
```

Apparently, the file size and mime type are not accepted by the media collection. Any ideas how to solve this?

**Comment by freekmurze** [comment]
I'd be open for a PR that handles this.

**Reply by jaap** [reply]
I have investigated this behaviour (#3735) a bit and came to the conclusion that I think it should not be part of this package.
It might be worth exploring if it should be 'fixed' in `Illuminate\Http\UploadedFile` itself.

I did find out that, by supplying an actual audio file to the UploadedFile it does propagate the MIME correctly.
It sort of defeats the purpose of UploadedFile, but I find it more readable/understandable that we're faking things rather than directly uploading the file contents.

So, in the end, your test could look something like this;

```php
    public function testItCascadesOnDelete()
    {
        Storage::fake('public');

        $project = Project::factory()->create();
        $track = ProjectTrack::factory()->create([
            'project_id' => $project->id,
        ]);

        $fakeFilePath = base_path('tests/data/test.mp3');
        $uploadFile = UploadedFile::fake()->createWithContent('test.mp3', file_get_contents($fakeFilePath));

        $media = $track->addMedia($uploadFile)
            ->toMediaCollection('track');

        // All is in the DB
        $this->assertDatabaseCount('media', 1);
        $this->assertDatabaseCount('project_tracks', 1);

        // And on disk
        $filePath = $media->getPath();
        $this->assertFileExists($filePath);

        // When we delete the project
        $project->delete();
        // The media is deleted trough cascading
        $this->assertDatabaseCount('project_tracks', 0);
        $this->assertDatabaseEmpty('media');

        // And the file is deleted
        $this->assertFileDoesNotExist($filePath);
    }
```
